sympy/stats/rv.py

```diff
@@ -586,14 +586,16 @@ def given(expr, condition=None, **kwargs):
     >>> X = Normal('X', 0, 1)
     >>> Y = Normal('Y', 0, 1)
     >>> pprint(density(X + Y, Y)(z), use_unicode=False)
-                    2
-           -(-Y + z)
-           -----------
-      ___       2
-    \/ 2 *e
-    ------------------
-             ____
-         2*\/ pi
+                           2       2
```

Comment:
This expression does not look like a density. I think that a conditional density, given `Y`, should also depend on `Y`.

Reply:
Yes you are right. I will resolve this. I did a mistake in analyzing.

Reply:
I agree that the expression should be simplified.

Reply:
I am thinking to use MGF in the case of normal distributions. Since, Normal distributions are completely determined by their second moment. We can generalize it for X1 + X2 + ... + Xn.

Reply:
I discussed with @Upabjojr and agree with him that ad-hoc techniques like, using MGF won't be a good idea for computer algebra systems. So, @jksuom if you have any generalised technique to suggest to simplify the result then it would be great.

Reply:
Simplification should not be handled by the _stats_ module. The _stats_ module should just compute things correctly. There is a separate _simplify_ module for simplifications.

Reply:
Yes, I agree for that too, it will make the code slower also as you suggested. `DiracDelta(Z - f(X, Y))` is fine for correctness. Maybe using some pre-existing results is the best approach, I can think as of now.